failed: ConnectionPoolSpec

<a href="https://github.com/patriknw"><img src="https://avatars.githubusercontent.com/u/336161?v=3" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [patriknw](https://github.com/patriknw)**
_Friday Aug 19, 2016 at 09:37 GMT_
_Originally opened as https://github.com/akka/akka/issues/21231_

---

https://jenkins.akka.io:8498/job/pr-validator-per-commit-jenkins/7117/consoleFull

```
[info] - should never close hot connections when minConnections key is given and >0 (minConnections = 5) *** FAILED *** (3 seconds, 117 milliseconds)
[info]   java.lang.AssertionError: assertion failed: timeout (3 seconds) during expectMsgClass waiting for class akka.stream.testkit.TestSubscriber$OnNext
[info]   at scala.Predef$.assert(Predef.scala:170)
[info]   at akka.testkit.TestKitBase$class.expectMsgClass_internal(TestKit.scala:435)
[info]   at akka.testkit.TestKitBase$class.expectMsgType(TestKit.scala:408)
[info]   at akka.testkit.TestKit.expectMsgType(TestKit.scala:737)
[info]   at akka.stream.testkit.TestSubscriber$ManualProbe.expectNextN(StreamTestKit.scala:317)
[info]   at akka.http.impl.engine.client.ConnectionPoolSpec$$anonfun$1$$anonfun$apply$mcV$sp$12$$anon$15.<init>(ConnectionPoolSpec.scala:267)
[info]   at akka.http.impl.engine.client.ConnectionPoolSpec$$anonfun$1$$anonfun$apply$mcV$sp$12.apply$mcV$sp(ConnectionPoolSpec.scala:252)
[info]   at akka.http.impl.engine.client.ConnectionPoolSpec$$anonfun$1$$anonfun$apply$mcV$sp$12.apply(ConnectionPoolSpec.scala:252)
[info]   at akka.http.impl.engine.client.ConnectionPoolSpec$$anonfun$1$$anonfun$apply$mcV$sp$12.apply(ConnectionPoolSpec.scala:252)
```

--- a/akka-http-core/src/test/scala/akka/http/impl/engine/client/ConnectionPoolSpec.scala
+++ b/akka-http-core/src/test/scala/akka/http/impl/engine/client/ConnectionPoolSpec.scala
@@ -311,36 +311,37 @@ class ConnectionPoolSpec extends AkkaSpec("""
 
       // for lower bound of one connection
       val minConnection = 1
-      val (requestIn, requestOut, responseOutSub, hcpMinConnection) =
+      val (requestIn, responseOut, responseOutSub, hcpMinConnection) =
         cachedHostConnectionPool[Int](idleTimeout = 100.millis, minConnections = minConnection)
       val gatewayConnection = hcpMinConnection.gateway
 
       acceptIncomingConnection()
       requestIn.sendNext(HttpRequest(uri = "/minimumslots/1", headers = immutable.Seq(close)) → 42)
       responseOutSub.request(1)
-      requestOut.expectNextN(1)
+      responseOut.expectNextN(1)
 
       condHolds(500.millis.dilated) { () ⇒
         Await.result(gatewayConnection.poolStatus(), 100.millis.dilated).get shouldBe a[PoolInterfaceRunning]
       }
     }
 
-    "never close hot connections when minConnections key is given and >0 (minConnections = 5)" in new TestSetup() {
+    "never close hot connections when minConnections key is given and >0 (minConnections = 5)" in new TestSetup(autoAccept = true) {
       val close: HttpHeader = Connection("close")
 
       // for lower bound of five connections
       val minConnections = 5
-      val (requestIn, requestOut, responseOutSub, hcpMinConnection) = cachedHostConnectionPool[Int](
+      val (requestIn, responseOut, responseOutSub, hcpMinConnection) = cachedHostConnectionPool[Int](
         idleTimeout = 100.millis,
         minConnections = minConnections,
         maxConnections = minConnections + 10)
 
-      (0 until minConnections) foreach { _ ⇒ acceptIncomingConnection() }
-      (0 until minConnections) foreach { i ⇒
-        requestIn.sendNext(HttpRequest(uri = s"/minimumslots/5/$i", headers = immutable.Seq(close)) → 42)
+      (1 to 30) foreach { _ ⇒ // run a few requests
+        (0 until minConnections) foreach { i ⇒
+          requestIn.sendNext(HttpRequest(uri = s"/minimumslots/5/$i", headers = immutable.Seq(close)) → 42)
+        }
+        responseOutSub.request(minConnections)
+        responseOut.expectNextN(minConnections)
       }
-      responseOutSub.request(minConnections)
-      requestOut.expectNextN(minConnections)
 
       val gatewayConnections = hcpMinConnection.gateway
       condHolds(1000.millis.dilated) { () ⇒

